### PR TITLE
Added parameter validation to PyObject methods

### PR DESCRIPTION
**Pull request by lostmsu**
### What does this implement/fix? Explain your changes.

This adds basic validations to input parameters of various `PyObject` methods, which should help in debugging code, that uses Python.NET.

Also adds a few function overloads

### Does this close any currently open issues?

No, but it reduces divergence with Lost Tech fork

### Checklist

Check all those that are applicable and complete.

-   [x] Make sure to include one or more tests for your change
-   [x] If an enhancement PR, please create docs and at best an example
-   [x] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by lostmsu** [comment]
@filmor any chance to look at this? This is a pretty light change.

File: src/runtime/pyobject.cs [hunk 1]
```diff
@@ -257,17 +268,41 @@ public PyObject GetAttr(string name)
...
+                Runtime.PyErr_Clear();
+                return _default;
+            }
+            return new PyObject(op);
+        }
+
+        /// <summary>
+        /// Gets the specified attribute. If attribute is not found, returns the fallback value.
+        /// </summary>
+        public PyObject GetAttrOrElse(string name, PyObject _default)
```
Comment (filmor) [inline src/runtime/pyobject.cs]: How much would it hurt to make this behaviour the default? We are using `GetAttr` a lot, I think (even in this file).

**Comment by filmor** [comment]
Almost completely fine, but I'm not quite sure about the `GetAttr` changes, in particular since `GetAttr(PyObject)` is unchanged AFAICT. Will have a closer look.

File: src/runtime/pyobject.cs (hunk 1 above)
Comment (lostmsu) [reply src/runtime/pyobject.cs]: @filmor I don't know, I don't have any legacy code :)
But that would be a breaking change in contract.

Comment (lostmsu) [reply src/runtime/pyobject.cs]: I decided to take this change out, so the rest could be merged.

**Comment by lostmsu** [comment]
@filmor I decided to take the `GetAttrOrElse` out for now so that the rest could be merged.

**Pull request merged**